### PR TITLE
Run away from getfuncargvalue deprecation.

### PR DESCRIPTION
**Pull request by amateja**
Howdy!

Here you have a patch for deprecated getfuncargvalue method. This PR is based on pytest-dev/pytest-django#393.

Cheers!

### CODE REVIEW

**Comment by amateja** [comment]
I know... tests are failing but it's due to PEP8 violation and version conflicts which is out of scope of this PR. Should fix for it be included too or you prefer separate PR?

**Comment by nicoddemus** [comment]
Thanks @amateja for the PR.

I'm not sure what pytest version is supposed to be supported by pytest-flask, but if it is >= 3.0 `request.getfixturevalue` can be used directly instead.

File: pytest_flask/pytest_compat.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+def getfixturevalue(request, value):
+    if hasattr(request, 'getfixturevalue'):
+        return request.getfixturevalue(value)
+
+    return request(value)
```
Comment (nicoddemus) [inline pytest_flask/pytest_compat.py]: Shouldn't this be `return request.getfuncargvalue(value)`?

**Comment by nicoddemus** [comment]
About if it is OK to fix the build on this PR or in another one, the maintainers must decide. 😁

File: pytest_flask/pytest_compat.py (hunk 1 above)
Comment (amateja) [inline pytest_flask/pytest_compat.py]: Nice catch! By accident I overwritten that line in 8aba5e0 before committing my changes and didn't fix it properly in 0c92db3. Sorry.

**Comment by amateja** [comment]
I'm not sure if this is fixable easily. Some package in dependency chain requires pytest>=2.7.0 and we are testing against 2.6.x. We have two options - drop testing against 2.6.x (which is not an direct equivalent of dropping support) or track down that dependency and pin its version in tox deps matrix.

**Comment by amateja** [comment]
Found it! It is because of recent changes in pytest-xdist. Will try to propose a workaround.

**Comment by nicoddemus** [comment]
IMHO dropping support for pytest < 2.7.0 is reasonable, 2.6.4 was released in Oct 24, 2014.

**Comment by amateja** [comment]
Actually dropping support for pytest 2.6.x might be reasonable. According to [pytest changelog](https://docs.pytest.org/en/latest/changelog.html#id170) python 3.5+ is supported only by pytest 2.7.3+. Keeping in mind that python 2.6 is deprecated and 3.3 is  going to be deprecated [this year](https://docs.python.org/devguide/index.html#status-of-python-branches) dependency matrix is going to be not that beautiful. Moreover pytest 3.1 was released couple days ago so we might it to test suite.

**Comment by vitalk** [comment]
Hi, @amateja!

Thank you so much for your contribution!

**Comment by remcohaszing** [comment]
@nicoddemus  Could you release this on pypi?

**Comment by vitalk** [comment]
@remcohaszing I publish a new release on this weekend. Thank you for patience.

**Comment by YunLiaw** [comment]
@vitalk I found that on [pypi](https://pypi.python.org/pypi/pytest-flask) that the latest version published is 0.10.0, which seems to be not the one with this MR. Could you help to make a new release?

**Comment by nicoddemus** [comment]
@vitalk could you also add me as maintainer in PyPI? Thanks!

**Pull request merged**